### PR TITLE
Allow legacy EDAM mappings.

### PR DESCRIPTION
**Pull request by jmchilton**
Implement legacy mapping files for override EDAM topics and operations for tool IDs.

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+from typing import (
+    cast,
+    Dict,
+    List,
+    NamedTuple,
+    Optional,
+    Tuple,
+)
+
+from pkg_resources import resource_string
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]: Can you use `importlib_resources` for this ?
https://importlib-resources.readthedocs.io/en/latest/migration.html#pkg-resources-resource-string

**Review by mvdbeek — APPROVED** [review]
Very cool, thank you @jmchilton. Any chance we could map this to our own ontology instead of EDAM (or to multiple ontologies), so we can replace the existing tool panel sections exactly as they are now ?

**Comment by bgruening** [comment]
@jmchilton we need a rebase here I think.

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    List,
+    NamedTuple,
+    Optional,
+    Tuple,
+)
+
+import importlib_resources
+
+from galaxy.tool_util.biotools import BiotoolsMetadataSource
+from galaxy.tool_util.parser import ToolSource
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]:
```suggestion
from galaxy.tool_util.biotools import BiotoolsMetadataSource
from galaxy.tool_util.parser import ToolSource
from galaxy.util.resources import files
```

We have a little wrapper there that deals with importlib.resources not being in python 3.9

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    for x in content.splitlines():
+        if x.startswith("#"):
+            continue
+        key, value = cast(Tuple[str, str], tuple(x.split("\t")))
+        mapping.setdefault(key, []).append(value)
+    return mapping
+
+
+BIOTOOLS_MAPPING_CONTENT = (
+    importlib_resources.files("galaxy.tool_util.ontologies")
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]:
```suggestion
   files("galaxy.tool_util.ontologies")
```

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+)
+BIOTOOLS_MAPPING: Dict[str, str] = dict(
+    [
+        cast(Tuple[str, str], tuple(x.split("\t")))
+        for x in BIOTOOLS_MAPPING_CONTENT.splitlines()
+        if not x.startswith("#")
+    ]
+)
+EDAM_OPERATION_MAPPING_CONTENT = (
+    importlib_resources.files("galaxy.tool_util.ontologies")
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]:
```suggestion
    files("galaxy.tool_util.ontologies")
```

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    .joinpath("edam_operation_mappings.tsv")
+    .read_bytes()
+    .decode("UTF-8")
+)
+EDAM_OPERATION_MAPPING: Dict[str, List[str]] = multi_dict_mapping(EDAM_OPERATION_MAPPING_CONTENT)
+
+EDAM_TOPIC_MAPPING_CONTENT = (
+    importlib_resources.files("galaxy.tool_util.ontologies")
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]:
```suggestion
    files("galaxy.tool_util.ontologies")
```

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    .joinpath("edam_topic_mappings.tsv")
+    .read_bytes()
+    .decode("UTF-8")
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]:
```suggestion
    .read_text()
```

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        cast(Tuple[str, str], tuple(x.split("\t")))
+        for x in BIOTOOLS_MAPPING_CONTENT.splitlines()
+        if not x.startswith("#")
+    ]
+)
+EDAM_OPERATION_MAPPING_CONTENT = (
+    importlib_resources.files("galaxy.tool_util.ontologies")
+    .joinpath("edam_operation_mappings.tsv")
+    .read_bytes()
+    .decode("UTF-8")
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]:
```suggestion
    .read_text()
```

File: lib/galaxy/tool_util/ontologies/ontology_data.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        key, value = cast(Tuple[str, str], tuple(x.split("\t")))
+        mapping.setdefault(key, []).append(value)
+    return mapping
+
+
+BIOTOOLS_MAPPING_CONTENT = (
+    importlib_resources.files("galaxy.tool_util.ontologies")
+    .joinpath("biotools_mappings.tsv")
+    .read_bytes()
+    .decode("UTF-8")
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/ontologies/ontology_data.py]:
```suggestion
    .read_text()
```

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @jmchilton!

**Pull request merged**